compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2697,10 +2698,27 @@ class Typer extends Namer
       case tree: Closure => cpy.Closure(tree)(tpt = TypeTree(pt)).withType(pt)
     }
 
+    def hasWildcardArg(tp: Type): Boolean = tp match {
+      case tp: AppliedType => tp.hasWildcardArg
+      case tp: AndOrType => hasWildcardArg(tp.tp1) || hasWildcardArg(tp.tp2)
+      case _: SingletonType | _: HKTypeLambda => false
+      case tp: TypeProxy => hasWildcardArg(tp.superType)
+      case _ => false
+    }
+
     def adaptToSubType(wtp: Type): Tree = {
       // try converting a constant to the target type
       val folded = ConstFold(tree, pt)
-      if (folded ne tree) return adaptConstant(folded, folded.tpe.asInstanceOf[ConstantType])
+      if (folded ne tree)
+        return adaptConstant(folded, folded.tpe.asInstanceOf[ConstantType])
+
+      // If tree's type is not stable and has wildcard arguments, make it
+      // stable by casting it to a skolem. I.e. a tree `t` of widened type `T`
+      // gets widened to `t.$asInstanceOf$[SkolemType(T)]`. This will enable
+      // capture comversion in the subsequent subtype check.
```

Comment:
typo:
```suggestion
      // capture conversion in the subsequent subtype check.
```

Reply:
Also I'm a bit confused on terminology: isn't "capture conversion" just another name for "skolemization" ? In that case the capture happens here, not in the subsequent subtype check.

Reply:
I now explain what I mean by capture conversion in `isSubArgs`. I am not sure it's the best term for what happens there, but it corresponds to what Java uses.